### PR TITLE
Redact header (from okhttp logging)

### PR DESCRIPTION
**Pull request by alorma**
Hi.

I've ported this feature from OkHttp Logging Interceptor:

https://github.com/square/okhttp/tree/master/okhttp-logging-interceptor

Now, there's an option to **redact** / **remove** a specific header content and replace by an empty string, so sensitive information is not set on the database

### CODE REVIEW

File: library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java [hunk 1]
```diff
@@ -73,6 +80,14 @@ public ChuckInterceptor maxContentLength(long max) {
         return this;
     }
 
+    public ChuckInterceptor redactHeader(String name) {
+        Set<String> newHeadersToRedact = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+        newHeadersToRedact.addAll(headersToRedact);
+        newHeadersToRedact.add(name);
+        headersToRedact = newHeadersToRedact;
+        return this;
```
Comment (olivierperez) [inline library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java]: Why don't you just `headersToRedact.add(name)`?

File: library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java [hunk 2]
```diff
@@ -73,6 +80,14 @@ public ChuckInterceptor maxContentLength(long max) {
         return this;
     }
 
+    public ChuckInterceptor redactHeader(String name) {
```
Comment (olivierperez) [inline library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java]: Some developers may need to redact multiple headers, we should have this method instead:
`public ChuckInterceptor redactHeaders(String names...) {`

File: library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java (hunk 1 above)
Comment (alorma) [reply library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java]: Done

File: library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java (hunk 2 above)
Comment (alorma) [reply library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java]: Done

File: library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java (hunk 1 above)
Comment (olivierperez) [reply library/src/main/java/com/readystatesoftware/chuck/api/ChuckInterceptor.java]: Add it to **library-no-op** too please :wink:

**Comment by alorma** [comment]
rebased against developm fixed `gradle.properties` conflict

**Pull request merged**